File: app/src/main/java/com/aar/app/wsp/data/room/AnswerLineConverter.java
package com.aar.app.wsp.data.room;

import androidx.room.TypeConverter;

import com.aar.app.wsp.model.UsedWord;

public final class AnswerLineConverter {

    @TypeConverter
    public static String answerLineToString(UsedWord.AnswerLine answerLine) {
        return answerLine == null ? null : answerLine.toString();
    }

    @TypeConverter
    public static UsedWord.AnswerLine stringToAnswerLine(String answerLineData) {
        if (answerLineData == null) return null;
        UsedWord.AnswerLine answerLine = new UsedWord.AnswerLine();
        answerLine.fromString(answerLineData);
        return answerLine;
    }
}

File: app/src/main/java/com/aar/app/wsp/data/room/AnswerLineConverter.kt
package com.aar.app.wsp.data.room

import androidx.room.TypeConverter
import com.aar.app.wsp.model.UsedWord.AnswerLine

object AnswerLineConverter {
    @JvmStatic
    @TypeConverter
    fun answerLineToString(answerLine: AnswerLine?): String? {
        return answerLine?.toString()
    }

    @JvmStatic
    @TypeConverter
    fun stringToAnswerLine(answerLineData: String?): AnswerLine? {
        if (answerLineData == null) return null
        val answerLine = AnswerLine()
        answerLine.fromString(answerLineData)
        return answerLine
    }
}